sw-emulator/lib/periph/src/mailbox.rs

```diff
@@ -86,154 +95,337 @@ impl Default for MailboxRam {
     }
 }
 
-#[derive(Clone)]
-pub struct Mailbox {
-    regs: Rc<RefCell<MailboxRegs>>,
+pub type Soc2CaliptraMailboxRegs = Rc<RefCell<MailboxRegs>>;
+pub fn soc2caliptra_mailbox_regs(sram: MailboxRam) -> Soc2CaliptraMailboxRegs {
+    Rc::new(RefCell::new(MailboxRegs::new(sram)))
 }
 
-impl Mailbox {
-    const OFFSET_LOCK: RvAddr = 0x00;
-    const OFFSET_CMD: RvAddr = 0x08;
-    const OFFSET_DLEN: RvAddr = 0x0C;
-    const OFFSET_DATAIN: RvAddr = 0x10;
-    const OFFSET_DATAOUT: RvAddr = 0x14;
-    const OFFSET_EXECUTE: RvAddr = 0x18;
-    const OFFSET_STATUS: RvAddr = 0x1C;
+#[derive(Clone)]
+pub struct MailboxExternal {
+    regs: Soc2CaliptraMailboxRegs,
+}
 
-    pub fn new(ram: MailboxRam) -> Self {
-        Self {
-            regs: Rc::new(RefCell::new(MailboxRegs::new(ram))),
-        }
+impl MailboxExternal {
+    pub fn new(regs: Soc2CaliptraMailboxRegs) -> Self {
+        Self { regs }
     }
-
     pub fn read_dlen(&mut self) -> Result<u32, BusError> {
```

Comment:
Who calls these MailboxExternal functions? If I delete them all the tests still pass.

Reply:
sure - will remove. Thanks!

Reply:
Done.